### DISCUSSION TITLE
(SHOW & TELL) Skyfield in QGIS

### DISCUSSION THREAD

**Discussion by hamiltoncj** [skyfielders/python-skyfield#621] [Show and tell]
I have partially implemented some of the Skyfield algorithms into a [QGIS](https://qgis.org/en/site/) plugin. Three of the algorithms show the coordinates where the sun, moon, and planets are directly overhead for a particular date and time. This is an example output.

![earth_moon_planets](https://user-images.githubusercontent.com/6978279/126550839-a43efc50-42f3-47f0-9ce3-fdd8a0839ba4.jpg)

Another panel allows the user to click on the map to capture the coordinate at that location and display sun and moon stats such as azimuth, elevation, moon phase, dawn, sunrise, solar noon, sunset, etc. This is an example.

![sun_lunar_info](https://user-images.githubusercontent.com/6978279/126551745-6a5416be-0a42-4c90-b0b1-0e761a151ae1.jpg)

If you want to try this in QGIS 3.14+ (don't use an older QGIS), you can download it as follows. If you are on windows and want to download a self contained windows plugin use this link to download:

[earthsunmoon-alpha-windows.zip](https://drive.google.com/file/d/1Pld4yIxBTTHA0FBwPbaUw_qfrIhHTawA/view?usp=sharing)

If you are using some other OS, use this link, but before you install it in QGIS open your OSGeo4W console and run "pip install skyfield'.

[earthsunmoon-alpha.zip](https://drive.google.com/file/d/1kJammSDWRTk7659Ncp3qAnNMQLggZ4Ks/view?usp=sharing)

In QGIS select the menu item "Plugins->Manage and Install Plugins..." Click on the "Install from ZIP" tab. Click on the "..." button and navigate to where you saved the plugin as a zip file and select it. Next click on the "Install Plugin" button. You will find the plugin installed in "Plugins->Earth, sun, moon, & planets". As I update this plugin, I will update this page.

Comments are welcome. I would like to have a day / night terminator algorithm as well, but the one suggested in an earlier discussion does not work correctly and I don't know how to fix it. I may start a new discussion on it.

**Comment by brandon-rhodes** [comment]
Thanks for sharing about your project! This is the first time I remember seeing Skyfield hooked up to QGIS.
